core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -145,6 +135,40 @@ internal class CacheFallbackOperations
             return users;
         }
 
+        public static Dictionary<String, AdalUserInfo> RemoveAdalUser(ILegacyCachePersistance legacyCachePersistance,
+            string displayableId, string environment, string identifier)
+        {
+            Dictionary<String, AdalUserInfo> users = new Dictionary<String, AdalUserInfo>();
+            try
+            {
+                IDictionary<AdalTokenCacheKey, AdalResultWrapper> dictionary =
+                    AdalCacheOperations.Deserialize(legacyCachePersistance.LoadCache());
+
+                List<AdalTokenCacheKey> keysToRemove = new List<AdalTokenCacheKey>();
+                foreach (KeyValuePair<AdalTokenCacheKey, AdalResultWrapper> pair in dictionary)
+                {
+                    if ( environment.Equals(new Uri(pair.Key.Authority).Host) &&
+                         displayableId.Equals(pair.Key.DisplayableId) &&
+                         identifier.Equals(ClientInfo.CreateFromJson(pair.Value.RawClientInfo).ToUserIdentifier()) )
+                    {
+                        keysToRemove.Add(pair.Key);
+                    }
+                }
+
+                foreach (AdalTokenCacheKey key in keysToRemove)
+                {
+                    dictionary.Remove(key);
+                }
+
+                legacyCachePersistance.WriteCache(AdalCacheOperations.Serialize(dictionary));
+            }
+            catch (Exception ex)
+            {
+                CoreLoggerBase.Default.Warning("RemoveAdalUser falid due to Exception - " + ex);
+            }
```

Comment:
spelling error: "failed"

Pii vs/ noPii logging needed here?

Reply:
good point on the pii logging given the ex is used (which will contain the stack trace which could potentially contain pii afaik).

---
In reply to: [201867787](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1126#discussion_r201867787) [](ancestors = 201867787)

Reply:
looks like all logging in core should be reviewed - all usage of CoreLoggerBase, it is used incorrectly - it is called only as pii or not pii - should be both, exception message is logged with non pii log methods

Reply:
@SomkaPe sounds good - I'll work on this later and then we can do a pass over core, adal, and msal
cc: @henrik-me